packages/builder-vite/envs.ts

```diff
@@ -22,7 +22,7 @@ export const allowedEnvPrefix = ['VITE_', 'STORYBOOK_'];
  * Customized version of stringifyProcessEnvs from @storybook/core-common which
  * uses import.meta.env instead of process.env and checks for allowed variables.
  */
-export function stringifyProcessEnvs(raw: EnvsRaw, envPrefix: UserConfig['envPrefix']) {
+export function stringifyProcessEnvs(raw: EnvsRaw, envPrefix: UserConfig['envPrefix'], isBuild: boolean) {
```

Comment:
Not sure what the preferred way to do this is, but happy to change it if needed.

Reply:
Let's use a `mode` of either `production` or `development`, to avoid a [boolean trap](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap).

Reply:
I feel like `mode` would be confusing here, because it's possible to build in `development` mode and serve in `production`. And `vite` itself uses `command` to derive this information: 
https://github.com/vitejs/vite/blob/908c9e4cdd2cceb0f01495e38066ffe33c21ddb8/packages/vite/src/node/plugins/define.ts#L12

Reply:
Would you be okay with `command: 'build' | 'serve'` as an argument?

Reply:
Sounds good, thanks.

Reply:
Done!